File: Behavioral/ChainOfResponsibility/Python/support_handler.py
# support_handler.py

from abc import ABC, abstractmethod

class SupportHandler(ABC):
  def __init__(self):
    self._next_handler = None

  def set_next_handler(self, handler):
    self._next_handler = handler

  @abstractmethod
  def handle_request(self, request):
    pass

File: Behavioral/ChainOfResponsibility/JavaScript/SupportHandler.js
// SupportHandler.js

class SupportHandler {
  constructor() {
    this.nextHandler = null;
  }

  setNextHandler(handler) {
    this.nextHandler = handler;
  }

  handleRequest(request) {
    throw new Error("handleRequest() must be implemented by subclasses");
  }
}

module.exports = SupportHandler;
